src/Core/src/Platform/iOS/KeyboardAutoManagerScroll.cs

```diff
@@ -0,0 +1,582 @@
+﻿/*
+ * This class is adapted from IQKeyboardManager which is an open-source
+ * library implemented for iOS to handle Keyboard interactions with
+ * UITextFields/UITextViews. Link to their MIT License can be found here:
+ * https://github.com/hackiftekhar/IQKeyboardManager/blob/7399efb730eea084571b45a1a9b36a3a3c54c44f/LICENSE.md
+ */
+
+using System;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using CoreGraphics;
+using Foundation;
+using UIKit;
+
+namespace Microsoft.Maui.Platform;
+
+internal static class KeyboardAutoManagerScroll
+{
+	static UIScrollView? LastScrollView;
+	static CGPoint StartingContentOffset;
+	static UIEdgeInsets StartingScrollIndicatorInsets;
+	static UIEdgeInsets StartingContentInsets;
+	static CGRect KeyboardFrame = CGRect.Empty;
+	static CGPoint TopViewBeginOrigin = new(nfloat.MaxValue, nfloat.MaxValue);
+	static readonly CGPoint InvalidPoint = new(nfloat.MaxValue, nfloat.MaxValue);
+	static double AnimationDuration = 0.25;
+	static UIView? View = null;
+	static UIView? ContainerView = null;
+	static CGRect? CursorRect = null;
+	internal static bool IsKeyboardShowing = false;
+	static int TextViewTopDistance = 20;
+	static int DebounceCount = 0;
+	static NSObject? WillShowToken = null;
+	static NSObject? WillHideToken = null;
+	static NSObject? TextFieldToken = null;
+	static NSObject? TextViewToken = null;
+
+	internal static void Init()
+	{
+		TextFieldToken = NSNotificationCenter.DefaultCenter.AddObserver(new NSString("UITextFieldTextDidBeginEditingNotification"), DidUITextBeginEditing);
```

Comment:
Is it worth checking here if it's already been `Init'd` ? 

If (TextfieldToken is not null)
    return;